framework/include/materials/CompositeTensorBase.h

```diff
@@ -106,12 +106,8 @@ CompositeTensorBase<T, U>::validParams()
   InputParameters params = U::validParams();
   params.addRequiredParam<std::vector<MaterialPropertyName>>("tensors", "Component tensors");
   params.addRequiredParam<std::vector<MaterialPropertyName>>("weights", "Component weights");
-  params.addDeprecatedCoupledVar("args",
-                                 "variable dependencies for the prefactor",
-                                 "args is deprecated, use 'coupled_variables' instead");
-  // TODO Make required once deprecation is handled, see #19119
-  params.addCoupledVar("coupled_variables", "variable dependencies for the prefactor");
-
+  params.addRequiredCoupledVar("args", "variable dependencies for the prefactor");
```

Comment:
eh doesnt matter

Reply:
The deprecation work piggy-backed off the rename work in which it is a prerequisite that the old parameter exist when the rename is attempted. Because we perform a map lookup for the old parameter, move its metadata over to the new parameter, and erase the old parameter.

For a developer performing deprecation, they will have already had this line in their code when they come to perform the deprecation. Then the deprecation is only a single new statement; they do not have to modify any statements. Of course when you finally remove the deprecated param you have to modify one statement, which modification could be avoided with your suggestion, and remove the other.

So the total developer work is the same ... but in terms of back-end implementation it is easier if I can reuse the same machinery for both renaming and deprecation